### DISCUSSION TITLE
Support for `.eln` files

### DISCUSSION THREAD

**Discussion by NicolasCARPi** [inveniosoftware/product-rdm#133] [Ideas]
Hello,

I'm writing to you as a member of [The ELN Consortium](https://the.elnconsortium.org/), a consortium of Electronic Laboratory Notebooks editors (+ other interested parties such as Dataverse or Indiscale).

We are working on a standard Research Data file format.

<details><summary>relevant xkcd</summary>

![standards](https://github.com/inveniosoftware/invenio-app-rdm/assets/3043706/acaf21ca-1234-4aa6-b361-566b299ff800)

</details>

So to avoid yet another standard we based our work on the already existing [RO-Crate](https://www.researchobject.org/ro-crate/1.1/) (Research Object Crate) and wrapped it up in a `.zip` archive.

I'm writing to you today to first let you know about this initiative, second ask you if you'd be interested in supporting a `.eln` in your previewer?

At the end of the day, it's a zip archive, so I don't know if you have a ZIP previewer already, and if it can easily be extended to consider `.eln` as a ZIP archive, too. (`.eln` is a standardized media type: https://www.iana.org/assignments/media-types/application/vnd.eln+zip).

The next step would be to allow actually reading the JSON-LD manifest file contained in there to enrich the metadata. I don't know if it would be a good idea to open the .zip and extract the files and store them separately, enriched by the metadata from the JSON-LD file describing the RO-Crate. That is up to discussion.

Please note that I have no knowledge of this app other than "this is what Zenodo uses".

If you can kindly point me to the relevant code parts, I'm willing to propose a PR for this, if that helps.

**Comment by tmorrell** [comment]
Thanks for stopping by! 

The previewer aspect should be an easy change - either add the extension to https://github.com/inveniosoftware/invenio-previewer/blob/master/invenio_previewer/extensions/zip.py or add a new extension .py file inspired by the zip one. The second option would be better if you want to adjust how the zip file is displayed (highlighting metadata or changing the standard zip previewer). Here is how the default zip previewer looks https://data.caltech.edu/records/rqsbv-j2j62

The metadata enrichment idea also sounds cool, but would be more involved. There are a couple of approaches...but the best would probably to identify the metadata when the file is uploaded and appropriately fill in the deposit form. That way users can see how the metadata extraction worked and modify it if necessary. This would require modifications to the uploader and deposit form, and would be a bit of a project. This issue is a good place to discuss and see if there is broader interest.

**Comment by NicolasCARPi** [comment]
Thank you for the pointers Tom :)

**Comment by SteffenBrinckmann** [comment]
I agree on the preview extension; since .eln is an RO-crate, does Zenodo plan to inherently have an ro-crate extension?

Regarding the rich metadata: a first step might be to use what is existing.  Zenodo has an API that can also handle the deposit form. If somebody from the .eln consortium has Zenodo support, then they could try to establish a mapping between ro-crate terms and zenodo terms. Then use upload in tandem with mapping. I will ping the .eln to see.
